File: src/bucles/while/while2.py
from typing import List, TypeVar
I = TypeVar(int)
def fib(n: int) -> str:
    a, b = 0, 1
    while a < n:
        print(a, end=' ')
        a, b = b, a + b
    print()
fib(1000)

#Utilizando un vector
def fibonacci(n: int) -> List[I]:
    array = []
    a = 0
    b = 1
    while a < n:
        a, b = b, b + a
        array.append(a)
    print(array)
fibonacci(1000)

File: src/bucles/while/while2.js
function fib(n) {
    let a = 0;
    let b = 1;
    do {
        console.log(a);
        let numerotemp = a;
        a = b;
        b = numerotemp + b;
    } while (a < n);
}
console.log(fib(n = 1000));


console.log('Hola');
//Version con while-loop
function fibw(n) {
    let a = 0;
    let b = 1;
    while (a < n) {
        console.log(a + ' ');
        let numtemp = a;
        a = b;
        b = numtemp + b;
    }
}
console.log(fibw(n = 1000));

console.log('hOLA ');
//Version arrow function
fibonacci = (n => {
    let a = 0;
    let b = 1;
    while (a < n) {
        console.log(a);
        let num = a;
        a = b;
        b = num + b;
    }
});
console.log(fibonacci(1000));

